src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonPropertyInfo.cs

```diff
@@ -448,19 +448,7 @@ internal void Configure()
 
             if (IsRequired)
             {
-                if (!CanDeserialize &&
```

Comment:
Could you keep the logic inlined please? Extracting into a separate method makes the diff harder to review and IMHO it shouldn't make a big difference from a readability perspective. Thanks.